numpy/core/tests/test_indexing.py

```diff
@@ -547,6 +547,17 @@ def test_object_assign(self):
         b[[0], :3] = [[1, (1,2), 3]]
         assert_array_equal(a, b[0])
 
+        # Check that swapping of axes works
+        b = b.T
+        b[:3, [0]] = [[1], [(1,2)], [3]]
+        assert_array_equal(a, b[:, 0])
+
+        # Another test for the memory order of the subspace
+        # (though it should not matter)
```

Comment:
Comparing the output the expected output now. Added the try/except for the first part. The second part actually never failed with an error, but checking the correctness is good anyway (though I admit, the more likely thing here is probably an error then incorrect result).